From 627ed3e522bd28d92a7d019b1ee9b6888a75e4dc Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 20 Jul 2018 16:25:24 +0800
Subject: [PATCH 4226/5242] MLK-18991-3: arm64:dts:imx8qm-mek: hdmi: Add hdmi
 in dts

commit  8ee144ae9b83496137b8ec1d25989add834b4bdb from
https://source.codeaurora.org/external/imx/linux-imx.git

Move hdmi in property to separate dts file.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 1e9df88385ca6045ec61737f88225c29ac36d68e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts  |   69 ++++++++++++++++++++
 .../boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts     |   50 --------------
 3 files changed, 70 insertions(+), 50 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index a16d118..c8d1284 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -24,6 +24,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-mek_ca53.dtb \
 				 fsl-imx8qm-mek_ca72.dtb \
 				 fsl-imx8qm-mek-hdmi.dtb \
+				 fsl-imx8qm-mek-hdmi-in.dtb \
 				 fsl-imx8qm-mek-dsi-rm67191.dtb \
 				 fsl-imx8qm-mek-enet2-tja1100.dtb \
 				 fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts
new file mode 100644
index 0000000..0daadcb
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/*
+ * HDMI IN only dts.
+ */
+
+#include "fsl-imx8qm-mek-hdmi.dts"
+
+/ {
+	sound-hdmi-rx {
+		compatible = "fsl,imx-audio-cdnhdmi";
+		model = "imx-audio-hdmi-rx";
+		audio-cpu = <&sai_hdmi_rx>;
+		protocol = <1>;
+		hdmi-in;
+	};
+};
+
+/* HDMI RX */
+&isi_0 {
+	status = "disabled";
+};
+
+&isi_1 {
+	interface = <4 0 2>;  /* <Input MIPI_VCx Output>
+						Input:  0-DC0, 1-DC1, 2-MIPI CSI0, 3-MIPI CSI1, 4-HDMI, 5-MEM
+						VCx:    0-VC0, 1-VC1, 2-VC2, 3-VC3, MIPI CSI only
+						Output: 0-DC0, 1-DC1, 2-MEM */
+	status = "okay";
+};
+
+&isi_2 {
+	status = "okay";
+	fsl,chain_buf;
+};
+
+&isi_3 {
+	status = "disabled";
+};
+
+&mipi_csi_0 {
+	status = "disabled";
+};
+
+&i2c0_mipi_csi0 {
+	status = "disabled";
+};
+
+&hdmi_rx {
+	fsl,cec;
+	assigned-clocks = <&clk IMX8QM_HDMI_RX_HD_REF_SEL>,
+						<&clk IMX8QM_HDMI_RX_PXL_SEL>,
+						<&clk IMX8QM_HDMI_RX_HD_REF_DIV>;
+	assigned-clock-parents = <&clk IMX8QM_HDMI_RX_DIG_PLL_CLK>,
+							<&clk IMX8QM_HDMI_RX_BYPASS_CLK>;
+	assigned-clock-rates = <0>, <0>, <400000000>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
index 6d94849..324ff44 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
@@ -27,14 +27,6 @@
 		hdmi-out;
 	};
 
-	sound-hdmi-rx {
-		compatible = "fsl,imx-audio-cdnhdmi";
-		model = "imx-audio-hdmi-rx";
-		audio-cpu = <&sai_hdmi_rx>;
-		protocol = <1>;
-		hdmi-in;
-	};
-
 	sound-amix-sai {
 		status = "disabled";
 	};
@@ -108,45 +100,3 @@
 	assigned-clock-rates = <786432000>, <49152000>, <12288000>;
 	status = "okay";
 };
-
-/* HDMI RX */
-&isi_0 {
-	status = "disabled";
-};
-
-&isi_1 {
-	interface = <4 0 2>;  /* <Input MIPI_VCx Output>
-						Input:  0-DC0, 1-DC1, 2-MIPI CSI0, 3-MIPI CSI1, 4-HDMI, 5-MEM
-						VCx:    0-VC0, 1-VC1, 2-VC2, 3-VC3, MIPI CSI only
-						Output: 0-DC0, 1-DC1, 2-MEM */
-	status = "okay";
-};
-
-&isi_2 {
-	status = "okay";
-	fsl,chain_buf;
-};
-
-&isi_3 {
-	status = "disabled";
-};
-
-
-&mipi_csi_0 {
-	status = "disabled";
-};
-
-&i2c0_mipi_csi0 {
-	status = "disabled";
-};
-
-&hdmi_rx {
-	fsl,cec;
-	assigned-clocks = <&clk IMX8QM_HDMI_RX_HD_REF_SEL>,
-						<&clk IMX8QM_HDMI_RX_PXL_SEL>,
-						<&clk IMX8QM_HDMI_RX_HD_REF_DIV>;
-	assigned-clock-parents = <&clk IMX8QM_HDMI_RX_DIG_PLL_CLK>,
-							<&clk IMX8QM_HDMI_RX_BYPASS_CLK>;
-	assigned-clock-rates = <0>, <0>, <400000000>;
-	status = "okay";
-};
-- 
1.7.9.5

